### PR TITLE
fixes #391

### LINKED ISSUE #391: Null ref parsing function inside object destructuring
When parsing functions inside an object destructuring the function will not have a binding and fail with a null ref exception.  I verified this in 1.21.4.

```
using NUglify.JavaScript;

new JSParser().Parse(@"
function trimConfig(config, reduce) {
   return {
      ...(config.paths && {
         paths: reduce(config.paths, (paths) => {
            return paths;
         }),
      }),
   };                                                                                                                                                                                                                                                      
}
");
```

Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at NUglify.JavaScript.Syntax.FunctionScope.DefineFunctionExpressionName()
   at NUglify.JavaScript.Syntax.FunctionScope.DeclareScope()
   at NUglify.JavaScript.Visitors.ResolutionVisitor.CreateFields(ActivationObject scope)
   at NUglify.JavaScript.Visitors.ResolutionVisitor.CreateFields(ActivationObject scope)
   at NUglify.JavaScript.Visitors.ResolutionVisitor.CreateFields(ActivationObject scope)
   at NUglify.JavaScript.Visitors.ResolutionVisitor.Apply(BlockStatement block, ActivationObject scope, JSParser parser)
   at NUglify.JavaScript.JSParser.InternalParse()
   at NUglify.JavaScript.JSParser.Parse(DocumentContext sourceContext)
   at NUglify.JavaScript.JSParser.Parse(String source)

### PR DESCRIPTION
**Pull request by timmcb**

### CODE REVIEW

**Comment by timmcb** [comment]
I forgot to check in the project file changes for the copy of the new test files.  Will fix asap.

**Comment by trullock** [comment]
Closing this due to the feedback, but ive cherrypicked the diff and will merge the fix in a second...

File: src/NUglify.Tests/JavaScript/Bugs.cs [hunk 1]
```diff
@@ -382,5 +382,11 @@ public void Bug375()
 	        TestHelper.Instance.RunTest("-rename:all");
         }
 
+        [Test]
+        public void Bug391()
+        {
+            TestHelper.Instance.RunErrorTest();
```
[the remark this answers is not in the document]
Comment (timmcb) [reply src/NUglify.Tests/JavaScript/Bugs.cs:388]: Ah, thanks.  A case of copy&paste without a good understanding of the infrastructure.  I guess the name could have clued me in :)

File: changelog.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## v1.21.5 (22 April 2024)
```
[the remark this answers is not in the document]
Comment (timmcb) [reply changelog.md:3]: No problem, next time I'll leave it out.

File: src/NUglify.Tests/JavaScript/Bugs.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (trullock) [reply src/NUglify.Tests/JavaScript/Bugs.cs:388]: No worries

File: changelog.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (trullock) [reply changelog.md:3]: Feel free to attack some more bugs ;)

**Pull request closed without merging**